.github/workflows/test.yml

```diff
@@ -18,66 +20,60 @@ permissions:
 jobs:
   lint:
     name: Lint
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
-        with:
-          cache: 'npm'
-          node-version: lts/*
-      - run: npm ci
-      - run: npm run check
+    uses: ./.github/workflows/reusable-npm-run.yml
+    with:
+      command: check
 
   test:
-    name: Node.js ${{ matrix.node-version }}
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        node-version:
-          - 22
-          - 20
-          - 18
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
-        with:
-          cache: 'npm'
-          node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run test-ci
+    name: Test on Node 18
+    uses: ./.github/workflows/reusable-npm-run.yml
+    with:
+      command: test-ci
+      node-version: 18
+
+  test-20:
+    name: Test on Node 20
+    if: github.event_name != 'schedule'
+    uses: ./.github/workflows/reusable-npm-run.yml
+    with:
+      command: test-ci
+      node-version: 20
+
+  test-22:
+    name: Test on Node 22
+    if: github.event_name != 'schedule'
+    uses: ./.github/workflows/reusable-npm-run.yml
+    with:
+      command: test-ci
+      node-version: 22
```

Comment:
I think that part can use a node version matrix to avoid code duplication. The condition might become a bit more complicated though.

Reply:
I do that in some other repositories, but that involves splitting a string into an array of strings through a two job process, and in this case I felt it was cleaner to just repeat the test three times – especially the runs becomes cleaner as they have no parse step

(Would be kind of neat to be able to send array data to a reusable workflow, but as far as I'm aware only strings and booleans can be sent)

Reply:
I guess another solution would be to pull them out into another reusable workflow and have that workflow use a matrix – so recursive reusable workflows, but that also feels a bit overkill

Reply:
```yaml
 test:
    name: Test on Node ${{ matrix.node }}
    uses: ./.github/workflows/reusable-npm-run.yml
    with:
      command: test-ci
      node-version: ${{ matrix.node }}
```

will not run it on all the node matrix?

Reply:
I was 100% sure that wasn't possible, maybe they added that recently: https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-a-matrix-strategy-with-a-reusable-workflow

I'll fix!

Reply:
Seems to work well 👍 Thanks for the suggestion @jerome-benoit!

Now I can remove the extra parsing step from all of my own reusable workflows https://github.com/voxpelli/ghatemplates/blob/0a20a5f68342d7c162637c56a856523592f4f982/.github/workflows/test.yml#L41-L51